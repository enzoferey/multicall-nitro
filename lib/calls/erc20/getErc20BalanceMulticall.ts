import type { Call } from "../../types";

export function getErc20BalanceMulticall(
  erc20ContractAddress: string,
  address: string,
  options: { label?: string } = {}
): Call {
  const { label = "balance" } = options;

  return {
    target: erc20ContractAddress,
    call: ["balanceOf(address)(uint256)", address],
    label,
  };
}
