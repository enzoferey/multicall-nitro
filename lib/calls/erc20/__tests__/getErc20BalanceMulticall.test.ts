import { describe, it, expect } from "vitest";

import { getErc20BalanceMulticall } from "../getErc20BalanceMulticall";

describe("getErc20BalanceMulticall", () => {
  it("should match the call", () => {
    expect(
      getErc20BalanceMulticall(
        "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
        "0xffffffffffffffffffffffffffffffffffffffff"
      )
    ).toMatchSnapshot();
  });
  it("should allow overwritting the value label", () => {
    expect(
      getErc20BalanceMulticall(
        "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
        "0xffffffffffffffffffffffffffffffffffffffff",
        { label: "some label" }
      ).label
    ).toMatchSnapshot();
  });
});
