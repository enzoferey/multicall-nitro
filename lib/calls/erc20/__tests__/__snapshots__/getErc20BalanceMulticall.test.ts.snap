// Vitest Snapshot v1

exports[`getErc20BalanceMulticall > should allow overwritting the value label 1`] = `"some label"`;

exports[`getErc20BalanceMulticall > should match the call 1`] = `
{
  "call": [
    "balanceOf(address)(uint256)",
    "0xffffffffffffffffffffffffffffffffffffffff",
  ],
  "label": "balance",
  "target": "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
}
`;
